### PR TITLE
feat(DTFS2-5850):  Dashboard filtering and logic

### PR DESCRIPTION
**Pull request by Zainzzkk**
# Issue: 
* DTFS2-6125 - selecting all deal status filters crashed portal-api
* DTFS2-6126 - filters with whitespaces or issued or unissued cannot be removed if selected
* filtering was on or logic as compared to and

# Changes made:
* Changed filtering logic to and.
* Within the same field, such as deal status, or logic is used - such as draft OR submitted
* Fixed the inability of removing filters with whitespaces by removing whitespaces from comparison value
* fixed crash by selecting all status filters

### CODE REVIEW

File: e2e-tests/portal/cypress/e2e/journeys/dashboard/dashboard-deals/dashboard-deals-filter-by-multiple-fields-multiple-values.spec.js [hunk 1]
```diff
@@ -154,9 +154,9 @@ context('Dashboard Deals filters - filter by multiple fields with multiple value
 
   it('renders only deals that have matching fields - MIA, AIN, Draft status, Ready for check status', () => {
     const EXPECTED_DEALS = ALL_DEALS.filter(({ submissionType, status }) =>
-      status === CONSTANTS.DEALS.DEAL_STATUS.DRAFT
-      || submissionType === CONSTANTS.DEALS.SUBMISSION_TYPE.MIA
-      || submissionType === CONSTANTS.DEALS.SUBMISSION_TYPE.AIN);
+      (status === CONSTANTS.DEALS.DEAL_STATUS.DRAFT || status === CONSTANTS.DEALS.DEAL_STATUS.READY_FOR_APPROVAL)
+      && (submissionType === CONSTANTS.DEALS.SUBMISSION_TYPE.MIA
+      || submissionType === CONSTANTS.DEALS.SUBMISSION_TYPE.AIN));
```
Comment (abhi-markan) [inline e2e-tests/portal/cypress/e2e/journeys/dashboard/dashboard-deals/dashboard-deals-filter-by-multiple-fields-multiple-values.spec.js:159]: Bring the condition to same line please.
Ideally 3 or more then each on new line.

File: e2e-tests/portal/cypress/e2e/journeys/dashboard/dashboard-facilities/dashboard-facilities-filter-by-multiple-fields-multiple-values.spec.js [hunk 2]
```diff
@@ -191,4 +189,56 @@ context('Dashboard Facilities filters - filter by multiple fields with multiple
...
+    dashboardFacilities.filters.panel.form.type.contingent.checkbox().click();
+    dashboardFacilities.filters.panel.form.submissionType.AIN.checkbox().click();
+    dashboardFacilities.filters.panel.form.submissionType.MIA.checkbox().click();
+    dashboardFacilities.filters.panel.form.submissionType.MIN.checkbox().click();
+    dashboardFacilities.filters.panel.form.hasBeenIssued.issued.checkbox().click();
+    dashboardFacilities.filters.panel.form.hasBeenIssued.unissued.checkbox().click();
+    filters.panel.form.applyFiltersButton().click();
+
+    const EXPECTED_FACILITIES = ALL_FACILITIES.filter(({ type, submissionType, hasBeenIssued }) =>
+      (submissionType === CONSTANTS.DEALS.SUBMISSION_TYPE.AIN || submissionType === CONSTANTS.DEALS.SUBMISSION_TYPE.MIA
```
Comment (abhi-markan) [inline e2e-tests/portal/cypress/e2e/journeys/dashboard/dashboard-facilities/dashboard-facilities-filter-by-multiple-fields-multiple-values.spec.js:234]: All conditions in separate line please.

File: e2e-tests/portal/cypress/e2e/journeys/dashboard/dashboard-deals/dashboard-deals-filter-by-multiple-fields-multiple-values.spec.js [hunk 3]
```diff
@@ -172,4 +172,78 @@ context('Dashboard Deals filters - filter by multiple fields with multiple value
...
+    dashboardDeals.filters.panel.form.submissionType.MIN.checkbox().click();
+    dashboardDeals.filters.panel.form.dealType.gef.checkbox().click();
+    dashboardDeals.filters.panel.form.dealType.bssEwcs.checkbox().click();
+
+    filters.panel.form.applyFiltersButton().click();
+
+    cy.url().should('eq', relative('/dashboard/deals/0'));
+
+    const EXPECTED_DEALS = ALL_DEALS.filter(({ dealType, status, submissionType }) =>
+      (submissionType === CONSTANTS.DEALS.SUBMISSION_TYPE.MIA
```
Comment (abhi-markan) [inline e2e-tests/portal/cypress/e2e/journeys/dashboard/dashboard-deals/dashboard-deals-filter-by-multiple-fields-multiple-values.spec.js:236]: Can we please add a case where filters are added and then removed.

**Review by abhi-markan — CHANGES_REQUESTED** [review]
Good work, few minor comments to address please.

File: e2e-tests/portal/cypress/e2e/journeys/dashboard/dashboard-deals/dashboard-deals-filter-by-multiple-fields-multiple-values.spec.js (hunk 1 above)
Comment (Zainzzkk) [reply e2e-tests/portal/cypress/e2e/journeys/dashboard/dashboard-deals/dashboard-deals-filter-by-multiple-fields-multiple-values.spec.js]: done

File: e2e-tests/portal/cypress/e2e/journeys/dashboard/dashboard-facilities/dashboard-facilities-filter-by-multiple-fields-multiple-values.spec.js (hunk 2 above)
Comment (Zainzzkk) [reply e2e-tests/portal/cypress/e2e/journeys/dashboard/dashboard-facilities/dashboard-facilities-filter-by-multiple-fields-multiple-values.spec.js]: done

File: e2e-tests/portal/cypress/e2e/journeys/dashboard/dashboard-deals/dashboard-deals-filter-by-multiple-fields-multiple-values.spec.js (hunk 3 above)
Comment (Zainzzkk) [reply e2e-tests/portal/cypress/e2e/journeys/dashboard/dashboard-deals/dashboard-deals-filter-by-multiple-fields-multiple-values.spec.js:235]: done in another test dashboard-facilities-main-container-filter-remove.spec and dashboard-deals-main-container-filter-remove.spec

**Review by abhi-markan — APPROVED** [review]

**Pull request merged**